crates/server/src/rpc/mod.rs

```diff
@@ -89,11 +70,20 @@ pub async fn run_rpc_server(
         })
         .recover(handle_rejection);
 
-    rpc_server.set_handler(|port| {
+    rpc_server.set_module_registrator_handler(|port| {
         // Registers service for every port
         QuestsServiceRegistration::register_service(port, QuestsServiceImplementation {})
     });
 
+    rpc_server.set_on_transport_closes_handler(move |_, transport_id| {
+        let subs_join_handle = subscriptions_handle_by_transport_id_cloned.clone();
+        let subs_generator = subscription_by_transport_id_cloned.clone();
+        tokio::spawn(async move {
```

Comment:
What are the criteria for spawning tasks?

Reply:
the closure doesn't accept returning an async block or a `Box::pin` so in order to be able to do async ops I use the `tokio::spawn`.